### PR TITLE
Rename institution to provider

### PR DESCRIPTION
**Pull request by MartinBentley**
### Context
This is part of the story to maintain consistency in naming conventions in the code.
Here is the card: https://trello.com/c/e6UKD5c2/541-rename-institution-provider-in-manage-courses.
Consumers of the Api will also need to be changed.
### Guidance to review
Good luck!

### CODE REVIEW

File: src/ManageCourses.Api/Controllers/EnrichmentController.cs [hunk 1]
```diff
@@ -18,62 +18,62 @@ public EnrichmentController(IEnrichmentService service)
...
         /// <returns>a data object. null if not found</returns>
         [HttpGet]
-        [Route("institution/{ucasInstitutionCode}")]
-        [ProducesResponseType(typeof(UcasInstitutionEnrichmentGetModel), 200)]
+        [Route("provider/{ucasProviderCode}")]
+        [ProducesResponseType(typeof(UcasProviderEnrichmentGetModel), 200)]
         [ProducesResponseType(400)]
         [ProducesResponseType(404)]
-        public ActionResult GetInstitution(string ucasInstitutionCode)
+        public ActionResult GetProvider(string ucasProviderCode)
```
Comment (defong) [inline src/ManageCourses.Api/Controllers/EnrichmentController.cs]: ucas?

File: src/ManageCourses.Api/Controllers/EnrichmentController.cs [hunk 2]
```diff
@@ -18,62 +18,62 @@ public EnrichmentController(IEnrichmentService service)
...
+        /// <param name="ucasProviderCode">provider code that relates to the Ucas provider</param>
         /// <param name="model">contains the payload that represents the data to be saved</param>
         [HttpPost]
-        [Route("institution/{ucasInstitutionCode}")]
+        [Route("provider/{ucasProviderCode}")]
         [ProducesResponseType(200)]
         [ProducesResponseType(400)]
         [ProducesResponseType(404)]
-        public ActionResult SaveInstitution(string ucasInstitutionCode, [FromBody] UcasInstitutionEnrichmentPostModel model)
+        public ActionResult SaveProvider(string ucasProviderCode, [FromBody] UcasProviderEnrichmentPostModel model)
```
Comment (defong) [inline src/ManageCourses.Api/Controllers/EnrichmentController.cs]: ucas?

File: src/ManageCourses.Api/Controllers/EnrichmentController.cs [hunk 3]
```diff
@@ -18,62 +18,62 @@ public EnrichmentController(IEnrichmentService service)
...
         /// <summary>
         /// always get the latest enrichment record regardless of status
         /// </summary>
-        /// <param name="ucasInstitutionCode">institution code that relates to the Ucas institution</param>
+        /// <param name="ucasProviderCode">provider code that relates to the Ucas provider</param>
         /// <param name="ucasCourseCode">Course code that relates to the Ucas Course of the enrichment data</param>
         /// <returns>a data object. null if not found</returns>
         [HttpGet]
-        [Route("institution/{ucasInstitutionCode}/course/{ucasCourseCode}")]
+        [Route("provider/{ucasProviderCode}/course/{ucasCourseCode}")]
```
Comment (defong) [inline src/ManageCourses.Api/Controllers/EnrichmentController.cs]: ucas?

File: src/ManageCourses.Api/Controllers/EnrichmentController.cs [hunk 4]
```diff
@@ -18,62 +18,62 @@ public EnrichmentController(IEnrichmentService service)
...
         /// <summary>
         /// saves a draft enrichment record (upsert)
         /// </summary>
-        /// <param name="ucasInstitutionCode"></param>
+        /// <param name="ucasProviderCode"></param>
         /// <param name="ucasCourseCode">Course code that relates to the Ucas Course</param>
         /// <param name="model">contains the payload that represents the data to be saved</param>
         [HttpPost]
-        [Route("institution/{ucasInstitutionCode}/course/{ucasCourseCode}")]
+        [Route("provider/{ucasProviderCode}/course/{ucasCourseCode}")]
```
Comment (defong) [inline src/ManageCourses.Api/Controllers/EnrichmentController.cs]: ucas?

File: src/ManageCourses.Api/Controllers/PublishController.cs [hunk 5]
```diff
@@ -69,31 +69,31 @@ public async Task<ActionResult> PublishCoursesToSearchAndCompare(string instCode
...
             }
 
             var courseMapper = new CourseMapper();
 
-            var ucasInstData = _dataService.GetUcasInstitutionForUser(name, instCode);
-            var orgEnrichmentData = _enrichmentservice.GetInstitutionEnrichment(instCode, name);
-            var ucasCourseData = _dataService.GetCourseForUser(name, instCode, courseCode);
-            var courseEnrichmentData = _enrichmentservice.GetCourseEnrichment(instCode, courseCode, name);
-            if (ucasInstData == null || ucasCourseData == null)
+            var ucasProviderData = _dataService.GetUcasProviderForUser(name, providerCode);
```
Comment (defong) [inline src/ManageCourses.Api/Controllers/PublishController.cs]: ucas?

File: src/ManageCourses.Api/Mapping/CourseMapper.cs [hunk 6]
```diff
@@ -14,22 +14,22 @@ public class CourseMapper : ICourseMapper
     {
         private readonly SubjectMapper subjectMapper = new SubjectMapper();
 
-        public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(Institution ucasInstData, Domain.Models.Course ucasCourseData, InstitutionEnrichmentModel institutionEnrichmentModel, CourseEnrichmentModel courseEnrichmentModel)
+        public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(Domain.Models.Provider ucasProviderData, Domain.Models.Course ucasCourseData, ProviderEnrichmentModel providerEnrichmentModel, CourseEnrichmentModel courseEnrichmentModel)
         {
-            ucasInstData = ucasInstData ?? new Institution();
+            ucasProviderData = ucasProviderData ?? new Domain.Models.Provider();
             ucasCourseData = ucasCourseData ?? new Domain.Models.Course();
             var sites = ucasCourseData.CourseSites ?? new ObservableCollection<CourseSite>();
```
Comment (defong) [inline src/ManageCourses.Api/Mapping/CourseMapper.cs]: Obervable Collection?

File: src/ManageCourses.Api/Mapping/CourseMapper.cs [hunk 7]
```diff
@@ -243,7 +243,7 @@ private string GetAccreditingInstitutionEnrichment(string accreditingInstCode, I
                 return "";
             }
 
-            var enrichment = enrichmentModel.AccreditingProviderEnrichments.FirstOrDefault(x => x.UcasInstitutionCode == accreditingInstCode);
+            var enrichment = enrichmentModel.AccreditingProviderEnrichments.FirstOrDefault(x => x.UcasProviderCode == accreditingProviderCode);
```
Comment (defong) [inline src/ManageCourses.Api/Mapping/CourseMapper.cs]: ucas?

File: src/ManageCourses.Api/Mapping/EnrichmentConverter.cs [hunk 8]
```diff
@@ -56,12 +56,12 @@ public UcasCourseEnrichmentGetModel Convert(CourseEnrichment source)
         /// </summary>
         /// <param name="source">enrichment data object</param>
         /// <returns>enrichment model with enrichment data (if found). Null if there is no source record</returns>
-        public UcasInstitutionEnrichmentGetModel Convert(InstitutionEnrichment source)
+        public UcasProviderEnrichmentGetModel Convert(ProviderEnrichment source)
```
Comment (defong) [inline src/ManageCourses.Api/Mapping/EnrichmentConverter.cs]: Ucas?

File: src/ManageCourses.Api/Mapping/ICourseMapper.cs [hunk 9]
```diff
@@ -5,6 +5,6 @@ namespace GovUk.Education.ManageCourses.Api.Mapping
 {
     public interface ICourseMapper
     {
-        SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(Institution ucasInstData, Course ucasCourseData, InstitutionEnrichmentModel orgEnrichmentModel, CourseEnrichmentModel courseEnrichmentModel);
+        SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(Provider ucasProviderData, Course ucasCourseData, ProviderEnrichmentModel providerEnrichmentModel, CourseEnrichmentModel courseEnrichmentModel);
```
Comment (defong) [inline src/ManageCourses.Api/Mapping/ICourseMapper.cs]: ucas?

File: src/ManageCourses.Api/Controllers/CoursesController.cs [hunk 10]
```diff
@@ -19,18 +16,18 @@ public CoursesController(IDataService dataService)
         }
 
         /// <summary>
-        /// Gets a course by institution Ucas code
+        /// Gets a course by provider Ucas code
```
Comment (defong) [inline src/ManageCourses.Api/Controllers/CoursesController.cs]: Ucas?

File: src/ManageCourses.Api/Controllers/CoursesController.cs [hunk 11]
```diff
@@ -42,24 +39,24 @@ public ActionResult Get(string instCode, string courseCode)
...
+        [Route("{providerCode}")]
         [ProducesResponseType(typeof(List<Course>), 200)]
         [ProducesResponseType(404)]
-        public ActionResult GetAll(string instCode)
+        public ActionResult GetAll(string providerCode)
         {
             var name = this.User.Identity.Name;
 
-            if (_dataService.GetUcasInstitutionForUser(name, instCode) == null)
+            if (_dataService.GetUcasProviderForUser(name, providerCode) == null)
```
Comment (defong) [inline src/ManageCourses.Api/Controllers/CoursesController.cs]: Ucas?

File: src/ManageCourses.Api/Controllers/OrganisationsController.cs [hunk 12]
```diff
@@ -19,18 +19,18 @@ public OrganisationsController(IDataService dataService)
         }
 
         /// <summary>
-        /// Gets an organisations by Institution Code
+        /// Gets an organisations by Provider Code
```
Comment (defong) [inline src/ManageCourses.Api/Controllers/OrganisationsController.cs]: Organisation?

File: src/ManageCourses.Api/Controllers/OrganisationsController.cs [hunk 13]
```diff
@@ -47,12 +47,12 @@ public ActionResult Get(string instCode)
         /// <returns>a list of UserOrganisation objects</returns>
```
Comment (defong) [inline src/ManageCourses.Api/Controllers/OrganisationsController.cs]: Comment?

File: src/ManageCourses.Api/Mapping/CourseMapper.cs [hunk 14]
```diff
@@ -14,22 +14,22 @@ public class CourseMapper : ICourseMapper
     {
         private readonly SubjectMapper subjectMapper = new SubjectMapper();
 
-        public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(Institution ucasInstData, Domain.Models.Course ucasCourseData, InstitutionEnrichmentModel institutionEnrichmentModel, CourseEnrichmentModel courseEnrichmentModel)
+        public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(Domain.Models.Provider ucasProviderData, Domain.Models.Course ucasCourseData, ProviderEnrichmentModel providerEnrichmentModel, CourseEnrichmentModel courseEnrichmentModel)
```
Comment (defong) [inline src/ManageCourses.Api/Mapping/CourseMapper.cs]: ucas?

File: src/ManageCourses.Api/Mapping/EnrichmentConverter.cs [hunk 15]
```diff
@@ -17,7 +17,7 @@ public EnrichmentConverter()
             };
         }        
 
-        public string ConvertToJson(UcasInstitutionEnrichmentPostModel model)
+        public string ConvertToJson(UcasProviderEnrichmentPostModel model)
```
Comment (defong) [inline src/ManageCourses.Api/Mapping/EnrichmentConverter.cs]: ucas?

File: src/ManageCourses.Api/Model/AccreditingProviderEnrichment.cs [hunk 16]
```diff
@@ -1,15 +1,15 @@
...
     /// These can be enriched. This is the storage of that enrichment.
     /// </summary>
     public class AccreditingProviderEnrichment
     {
         /// <summary>
-        /// Foreign key to the ucas institution that is this accrediting provider.
+        /// Foreign key to the ucas provider that is this accrediting provider.
         /// </summary>
-        public string UcasInstitutionCode { get; set; }
+        public string UcasProviderCode { get; set; }
```
Comment (defong) [inline src/ManageCourses.Api/Model/AccreditingProviderEnrichment.cs]: ucas?

File: src/ManageCourses.Api/Model/AccreditingProviderEnrichment.cs [hunk 17]
```diff
@@ -1,15 +1,15 @@
 ﻿namespace GovUk.Education.ManageCourses.Api.Model
 {
     /// <summary>
-    /// Institutions have courses that have accrediting_provider entries that are FKs to ucas_institution's
+    /// Providers have courses that have accrediting_provider entries that are FKs to ucas_providers's
```
Comment (defong) [inline src/ManageCourses.Api/Model/AccreditingProviderEnrichment.cs]: ucas?

File: src/ManageCourses.Api/Model/UcasProviderEnrichmentGetModel.cs [hunk 18]
```diff
@@ -3,13 +3,13 @@
 
 namespace GovUk.Education.ManageCourses.Api.Model
 {
-    public class UcasInstitutionEnrichmentGetModel
+    public class UcasProviderEnrichmentGetModel
```
Comment (defong) [inline src/ManageCourses.Api/Model/UcasProviderEnrichmentGetModel.cs]: ucas?

File: src/ManageCourses.Api/Model/UcasProviderEnrichmentPostModel.cs [hunk 19]
```diff
@@ -0,0 +1,11 @@
+namespace GovUk.Education.ManageCourses.Api.Model
+{
+    public class UcasProviderEnrichmentPostModel
```
Comment (defong) [inline src/ManageCourses.Api/Model/UcasProviderEnrichmentPostModel.cs]: ucas?

File: src/ManageCourses.Api/Controllers/EnrichmentController.cs [hunk 20]
```diff
@@ -18,62 +18,62 @@ public EnrichmentController(IEnrichmentService service)
         /// <summary>
         /// always get the latest enrichment record regardless of status
         /// </summary>
-        /// <param name="ucasInstitutionCode">institution code that relates to the Ucas institution of the enrichment data</param>
+        /// <param name="providerCode">provider code that relates to the provider of the enrichment data</param>
         /// <returns>a data object. null if not found</returns>
         [HttpGet]
-        [Route("institution/{ucasInstitutionCode}")]
-        [ProducesResponseType(typeof(UcasInstitutionEnrichmentGetModel), 200)]
+        [Route("provider/{providerCode}")]
```
Comment (timabell) [inline src/ManageCourses.Api/Controllers/EnrichmentController.cs]: keep the old route as well till all deployed, card up removal of old route

**Pull request merged**